poetry/installation/chooser.py

```diff
@@ -0,0 +1,180 @@
+import re
+
+from typing import List
+from typing import Tuple
+
+from packaging.tags import Tag
+from poetry.core.packages.package import Package
+from poetry.core.packages.utils.link import Link
+from poetry.repositories.pool import Pool
+from poetry.utils.env import Env
+from poetry.utils.patterns import wheel_file_re
+
+
+class InvalidWheelName(Exception):
+
+    pass
+
+
+class Wheel(object):
+    def __init__(self, filename):  # type: (str) -> None
+        wheel_info = wheel_file_re.match(filename)
+        if not wheel_info:
+            raise InvalidWheelName("{} is not a valid wheel filename.".format(filename))
```

Comment:
Do we expect this to be valid for local wheels as well? I am guessing right now we only expect this code path to be traversed only if we are installing from an index?

Reply:
The wheel files naming is standardized so it should be valid for local wheels too.